tests/test_metric.py

```diff
@@ -424,3 +429,178 @@ def _assemble_matched_predictions(system_id):
     assert matched_reference == reference
     assert matched_poses == poses
     return reference, poses
+
+
+def _test_mols_valence_violations() -> dict[Chem.Mol, int]:
+    good_smiles = [
+        "C1=CC=CC=C1",
+        "C1=CC=CC=C1[H]",
+        "N(=O)=O",
+        "C=P(=O)O",
+        "O=Cl(=O)O",
+        "OC1C(COP(=O)(OP(=O)(OP(=O)(O)O)O)O)OC(C1O)n1cnc2c1ncnc2N",
+        "CO(C)C",  # Oxygen charge +1
+    ]
+
+    one_viol = [
+        "C[C@H](FC)C",
+        "CF(C)C",
+    ]
+
+    two_viols = ["NC(O)NCCC1:C:C:C:C([C@H](C=F)/C=O/C(=O)(O)O):C:1"]
+
+    params = Chem.SmilesParserParams()
+    params.sanitize = False
+    good_mols = {Chem.MolFromSmiles(smile, params=params): 0 for smile in good_smiles}
+    one_viol = {Chem.MolFromSmiles(smile, params=params): 1 for smile in one_viol}
+    two_viols = {Chem.MolFromSmiles(smile, params=params): 2 for smile in two_viols}
+    return good_mols | one_viol | two_viols
+
+
+@pytest.mark.parametrize(
+    ["mol", "expected_viols"],
+    [
+        (mol, num_exp_viols)
+        for mol, num_exp_viols in _test_mols_valence_violations().items()
+    ],
+)
+def test_count_valence_violations(mol, expected_viols):
```

Comment:
I would be a but cautious testing private functions, because they require the reader to know more than the API and because private functions change more frequently than the relatively stable public API. I usually only resort to this, if some functionality cannot be reasonably tested using the public API.

In this case however, the test below achieves basically the same, as `LigandValenceViolations.evaluate()` is only a thin wrapper around `_count_valence_violations()`. Therefore I think we can remove this test.